hist/hist/src/HFitImpl.cxx

```diff
@@ -234,7 +234,11 @@ TFitResultPtr HFit::Fit(FitObject * h1, TF1 *f1 , Foption_t & fitOption , const
    // set the fit function
    // if option grad is specified use gradient
    if ( (linear || fitOption.Gradient) )
-      fitter->SetFunction(ROOT::Math::WrappedMultiTF1(*f1) );
+      fitter->SetFunction(ROOT::Math::WrappedMultiTF1(*f1));
+#ifdef R__HAS_VECCORE      
+   else if(f1->IsVectorized())
```

Comment:
Hm, could that `IsVectorized` be compile-time computable (eg. constexpr/non-type template parameter). This way we can have an `std::enable_if` based on that, doing the `SetFunction`. Once we have that perhaps this `#ifdef`s can go away.